spark/src/main/java/org/apache/spark/shuffle/sort/CometShuffleExternalSorter.java

```diff
@@ -257,6 +257,7 @@ private long getMemoryUsage() {
     for (SpillSorter sorter : spillingSorters) {
       totalPageSize += sorter.getMemoryUsage();
     }
+    totalPageSize += activeSpillSorter.getMemoryUsage();
```

Comment:
[doubt] any ideas for testing it ? whats the max variance this can cause in worst case ? activeSpillSorter is just full ?

Reply:
I'm not sure how to test this in a unit test, but this is clearly something missing. In the non-async path, only `activeSpillSorter` is used for sorting and spilling, and currently `getMemoryUsage` doesn't take account of it.